main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -2362,9 +2363,9 @@ object Resolver {
       // Case 0: No matches. Error.
       case Nil => Result.Err(ResolutionError.UndefinedStruct(qname, qname.loc))
       // Case 1: Exactly one match. Success.
-      case st :: _ => Result.Ok(st)
+      case st :: Nil => Result.Ok(st)
```

Comment:
Either we have discovered a bug in the compiler or the comments do not match what is actually intended here.